datavault-webapp/src/main/java/org/datavaultplatform/webapp/controllers/admin/AdminRoleAssignmentController.java

```diff
@@ -0,0 +1,136 @@
+            @Valid @NotNull @RequestParam("assignment") Long assignmentId,
+            @Valid RoleAssignmentRequest assignmentRequest) {
+
+        RoleAssignment roleAssignment = rest.getRoleAssignment(assignmentId)
+                .orElseThrow(() -> new EntityNotFoundException(RoleAssignment.class, String.valueOf(assignmentId)));
+
+        User user = rest.getUser(assignmentRequest.user);
+        RoleModel role = rest.getRole(assignmentRequest.role).orElseThrow
+                (()  -> new EntityNotFoundException(RoleModel.class, String.valueOf(assignmentRequest.role)));
+
+        roleAssignment.setUser(user);
+        roleAssignment.setRole(role);
+
+        return ResponseEntity.ok().build();
+    }
+
+    @PostMapping("/security/roles/{roleType}/{target}/user/delete")
+    @PreAuthorize("hasPermission(#targetId, #type, 'MANAGE_ROLES')")
+    public ResponseEntity removeRoleAssignment(@PathVariable("roleType") String type,
+                                               @PathVariable("target") String targetId,
+                                               @RequestParam("assignment") long assignmentId) {
+
+        RoleAssignment assignment = rest.getRoleAssignment(assignmentId)
+                .orElseThrow(() -> new EntityNotFoundException(RoleAssignment.class, String.valueOf(assignmentId)));
+
+        RoleModel role = assignment.getRole();
+
+        if (role.getAssignedUserCount() > 0) {
+            return ResponseEntity.status(HttpStatus.UNPROCESSABLE_ENTITY)
+                    .body("This role has users currently assigned");
+        }
+
+        rest.deleteRoleAssignment(assignment.getId());
+
+        return ResponseEntity.ok().build();
+    }
+
+    @PreAuthorize("hasPermission(#targetId, #type, 'MANAGE_ROLES')")
+    @PostMapping("/security/roles/{roleType}/{target}/user")
+    public ResponseEntity createRoleAssignment(
+            @PathVariable("roleType") String type,
+            @PathVariable("target") String targetId,
+            @RequestParam("role") long roleId,
+            @RequestParam("user") String userId) {
+
+        RoleAssignment assignment = new RoleAssignment();
+
+        RoleType roleType = RoleType.valueOf(type.toUpperCase());
+        switch (roleType) {
+            case SCHOOL:
+                assignment.setSchool(rest.getGroup(targetId));
+                break;
+            case VAULT:
+                assignment.setVault(rest.getActualVault(targetId));
+                break;
+            default:
+                return ResponseEntity.notFound().build();
+        }
+
+        User user = rest.getUser(userId);
+        if (user == null) {
+            throw new EntityNotFoundException(User.class, userId);
+        }
+
+        RoleModel role = rest.getRole(roleId)
+                .orElseThrow(() -> new EntityNotFoundException(RoleModel.class, String.valueOf(roleId)));
+
+        // @TODO - Check if RoleModel.roleType matches roleType if really necessary.
```

Comment:
I think we can probably drop the `roleType` request param altogether and just use the one from the role